### PR TITLE
Add support for tracking unstable edges

### PR DESCRIPTION
**Pull request by WorksButNotTested**
Add support for logging block ids and their associated ranges. As well as using the block id rather than the edge id when updating the map. This allows the log to be correlated with `var_bytes` in `fuzzer_stats` to allow the user to determine which address ranges in the target application resulted in the instability.

### CODE REVIEW

**Comment by vanhauser-thc** [comment]
this approach costs speed for every edge even if the feature is not enabled..
how about depending if unstable logging is on or off you either assign function a or function b?

**Comment by vanhauser-thc** [comment]
to clarify, if unstable logging is not enabled you assign afl_maybe_log for the instrumentation, otherwise you assign the function afl_maybe_log_unstable_logging (that you create). this way no speed decrease.

**Comment by WorksButNotTested** [comment]
So I split the function into two and move the branch up to here?
https://github.com/AFLplusplus/qemuafl/blob/3f18f6f0a34d071869a38a239fe5faedccbd89d7/accel/tcg/translate-all.c#L144

Then the overhead of the branch will occur at block translation time rather than block execution?

**Comment by vanhauser-thc** [comment]
yes that is the idea

**Comment by WorksButNotTested** [comment]
Does this look better? Think that should bring the branch back to the block translation stage (which shouldn't be running all that frequently).

File: accel/tcg/translate-all.c [hunk 1]
```diff
@@ -71,16 +71,33 @@
...
 
   INC_AFL_AREA(afl_idx);
 
   // afl_prev_loc = ((cur_loc & (MAP_SIZE - 1) >> 1)) |
   //                ((cur_loc & 1) << ((int)ceil(log2(MAP_SIZE)) -1));
   afl_prev_loc = cur_loc >> 1;
+}
 
+void HELPER(afl_maybe_log_trace)(target_ulong cur_loc) {
+  register uintptr_t afl_idx = cur_loc;
```
Comment (vanhauser-thc) [inline accel/tcg/translate-all.c:99]: I think you can just do `INC_AFL_AREA(cur_loc)`

Comment (WorksButNotTested) [reply accel/tcg/translate-all.c]: You are right! Is this better?

**Comment by vanhauser-thc** [comment]
lgtm :)

**Pull request merged**